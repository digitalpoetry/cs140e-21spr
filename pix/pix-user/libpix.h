#ifndef __LIBPIX_H__
#define __LIBPIX_H__
// support.
#include <stdint.h>
#include "syscalls.h"

#define die(x) do { putk(x); sys_exit(1); } while(0)

// invoke system call <sysno>
int syscall_invoke_asm(uint32_t sysno, ...);

#define sys_putc(x)     syscall_invoke_asm(SYS_PUTC, x)
#define sys_put_hex(x)  syscall_invoke_asm(SYS_PUT_HEX, x)
void sys_exit(int code);

// user level libc.
#include <string.h>
void putk(const char *msg);
void printk(const char *fmt, ...);

#endif
