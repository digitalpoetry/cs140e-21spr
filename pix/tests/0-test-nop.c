// just return: should be simple to pass!
#include "libpix.h"

void notmain(void) { return; }
