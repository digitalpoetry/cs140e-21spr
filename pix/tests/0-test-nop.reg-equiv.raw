FOUND: </dev/ttyUSB0>
opened tty port </dev/ttyUSB0>.
my-install: tty-usb=</dev/ttyUSB0> program=<./0-test-nop.ptag> about to boot
TRACE:simple_boot: sending 10942 bytes, crc32=f90cf6ab
waiting for a start
putting code
pi: <engler: CRC checked out!>
bootloader: Done.
listening on ttyusb=</dev/ttyUSB0>
kernel: stack is roughly at: 0x7ffffe8
user_code=0x400004, prog name=<0-test-nop.bin>
TRACE:	reg hash=0xfac07451
TRACE:	spsr=0x190
TRACE:	pc = 0x400004, lr = 0x400004
TRACE:	regs[0] = 0x400004
TRACE:	regs[1] = 0x7000000
TRACE:	regs[2] = 0x0
TRACE:	regs[3] = 0x0
TRACE:	regs[4] = 0x0
TRACE:	regs[5] = 0x0
TRACE:	regs[6] = 0x0
TRACE:	regs[7] = 0x0
TRACE:	regs[8] = 0x0
TRACE:	regs[9] = 0x0
TRACE:	regs[10] = 0x0
TRACE:	regs[11] = 0x0
TRACE:	regs[12] = 0x0
TRACE:	regs[13] = 0x7000000
TRACE:	regs[14] = 0x0
TRACE:	regs[15] = 0x400004
TRACE:------------------------------------------------------
TRACE:	reg hash=0x831b8654
TRACE:	spsr=0x190
TRACE:	pc = 0x400010, lr = 0x400010
TRACE:	regs[0] = 0x400004
TRACE:	regs[1] = 0x7000000
TRACE:	regs[2] = 0x0
TRACE:	regs[3] = 0x0
TRACE:	regs[4] = 0x0
TRACE:	regs[5] = 0x0
TRACE:	regs[6] = 0x0
TRACE:	regs[7] = 0x0
TRACE:	regs[8] = 0x0
TRACE:	regs[9] = 0x0
TRACE:	regs[10] = 0x0
TRACE:	regs[11] = 0x0
TRACE:	regs[12] = 0x0
TRACE:	regs[13] = 0x7000000
TRACE:	regs[14] = 0x400008
TRACE:	regs[15] = 0x400010
TRACE:------------------------------------------------------
TRACE:	reg hash=0xdb3c14aa
TRACE:	spsr=0x190
TRACE:	pc = 0x400008, lr = 0x400008
TRACE:	regs[0] = 0x400004
TRACE:	regs[1] = 0x7000000
TRACE:	regs[2] = 0x0
TRACE:	regs[3] = 0x0
TRACE:	regs[4] = 0x0
TRACE:	regs[5] = 0x0
TRACE:	regs[6] = 0x0
TRACE:	regs[7] = 0x0
TRACE:	regs[8] = 0x0
TRACE:	regs[9] = 0x0
TRACE:	regs[10] = 0x0
TRACE:	regs[11] = 0x0
TRACE:	regs[12] = 0x0
TRACE:	regs[13] = 0x7000000
TRACE:	regs[14] = 0x400008
TRACE:	regs[15] = 0x400008
TRACE:------------------------------------------------------
TRACE:	reg hash=0x79ec1afc
TRACE:	spsr=0x190
TRACE:	pc = 0x40000c, lr = 0x40000c
TRACE:	regs[0] = 0xffffffff
TRACE:	regs[1] = 0x7000000
TRACE:	regs[2] = 0x0
TRACE:	regs[3] = 0x0
TRACE:	regs[4] = 0x0
TRACE:	regs[5] = 0x0
TRACE:	regs[6] = 0x0
TRACE:	regs[7] = 0x0
TRACE:	regs[8] = 0x0
TRACE:	regs[9] = 0x0
TRACE:	regs[10] = 0x0
TRACE:	regs[11] = 0x0
TRACE:	regs[12] = 0x0
TRACE:	regs[13] = 0x7000000
TRACE:	regs[14] = 0x400008
TRACE:	regs[15] = 0x40000c
TRACE:------------------------------------------------------
0-test-nop.bin: sys_exit(-1): going to reboot
part=4
equiv values
TRACE:EQUIV:	number instructions = 10
TRACE:EQUIV:	reg hash = 0x28c48d80
DONE!!!

Saw done

bootloader: pi exited.  cleaning up
