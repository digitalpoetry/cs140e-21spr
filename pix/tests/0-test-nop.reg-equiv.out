TRACE:	reg hash=0xfac07451
TRACE:	spsr=0x190
TRACE:	pc = 0x400004, lr = 0x400004
TRACE:	regs[0] = 0x400004
TRACE:	regs[1] = 0x7000000
TRACE:	regs[2] = 0x0
TRACE:	regs[3] = 0x0
TRACE:	regs[4] = 0x0
TRACE:	regs[5] = 0x0
TRACE:	regs[6] = 0x0
TRACE:	regs[7] = 0x0
TRACE:	regs[8] = 0x0
TRACE:	regs[9] = 0x0
TRACE:	regs[10] = 0x0
TRACE:	regs[11] = 0x0
TRACE:	regs[12] = 0x0
TRACE:	regs[13] = 0x7000000
TRACE:	regs[14] = 0x0
TRACE:	regs[15] = 0x400004
TRACE:------------------------------------------------------
TRACE:	reg hash=0x831b8654
TRACE:	spsr=0x190
TRACE:	pc = 0x400010, lr = 0x400010
TRACE:	regs[0] = 0x400004
TRACE:	regs[1] = 0x7000000
TRACE:	regs[2] = 0x0
TRACE:	regs[3] = 0x0
TRACE:	regs[4] = 0x0
TRACE:	regs[5] = 0x0
TRACE:	regs[6] = 0x0
TRACE:	regs[7] = 0x0
TRACE:	regs[8] = 0x0
TRACE:	regs[9] = 0x0
TRACE:	regs[10] = 0x0
TRACE:	regs[11] = 0x0
TRACE:	regs[12] = 0x0
TRACE:	regs[13] = 0x7000000
TRACE:	regs[14] = 0x400008
TRACE:	regs[15] = 0x400010
TRACE:------------------------------------------------------
TRACE:	reg hash=0xdb3c14aa
TRACE:	spsr=0x190
TRACE:	pc = 0x400008, lr = 0x400008
TRACE:	regs[0] = 0x400004
TRACE:	regs[1] = 0x7000000
TRACE:	regs[2] = 0x0
TRACE:	regs[3] = 0x0
TRACE:	regs[4] = 0x0
TRACE:	regs[5] = 0x0
TRACE:	regs[6] = 0x0
TRACE:	regs[7] = 0x0
TRACE:	regs[8] = 0x0
TRACE:	regs[9] = 0x0
TRACE:	regs[10] = 0x0
TRACE:	regs[11] = 0x0
TRACE:	regs[12] = 0x0
TRACE:	regs[13] = 0x7000000
TRACE:	regs[14] = 0x400008
TRACE:	regs[15] = 0x400008
TRACE:------------------------------------------------------
TRACE:	reg hash=0x79ec1afc
TRACE:	spsr=0x190
TRACE:	pc = 0x40000c, lr = 0x40000c
TRACE:	regs[0] = 0xffffffff
TRACE:	regs[1] = 0x7000000
TRACE:	regs[2] = 0x0
TRACE:	regs[3] = 0x0
TRACE:	regs[4] = 0x0
TRACE:	regs[5] = 0x0
TRACE:	regs[6] = 0x0
TRACE:	regs[7] = 0x0
TRACE:	regs[8] = 0x0
TRACE:	regs[9] = 0x0
TRACE:	regs[10] = 0x0
TRACE:	regs[11] = 0x0
TRACE:	regs[12] = 0x0
TRACE:	regs[13] = 0x7000000
TRACE:	regs[14] = 0x400008
TRACE:	regs[15] = 0x40000c
TRACE:------------------------------------------------------
TRACE:EQUIV:	number instructions = 10
TRACE:EQUIV:	reg hash = 0x28c48d80
